XFree86-3.3/xc/programs/xinit/startx.man
.\" $XConsortium: startx.man /main/8 1996/12/09 17:09:15 kaleb $
.\" Copyright (c) 1993  X Consortium
.\" 
.\" Permission is hereby granted, free of charge, to any person obtaining
.\" a copy of this software and associated documentation files (the
.\" "Software"), to deal in the Software without restriction, including
.\" without limitation the rights to use, copy, modify, merge, publish,
.\" distribute, sublicense, and/or sell copies of the Software, and to
.\" permit persons to whom the Software is furnished to do so, subject to
.\" the following conditions:
.\" 
.\" The above copyright notice and this permission notice shall be included
.\" in all copies or substantial portions of the Software.
.\" 
.\" THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
.\" OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
.\" MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
.\" IN NO EVENT SHALL THE X CONSORTIUM BE LIABLE FOR ANY CLAIM, DAMAGES OR
.\" OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
.\" ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
.\" OTHER DEALINGS IN THE SOFTWARE.
.\" 
.\" Except as contained in this notice, the name of the X Consortium shall
.\" not be used in advertising or otherwise to promote the sale, use or
.\" other dealings in this Software without prior written authorization
.\" from the X Consortium.
.TH STARTX 1 "Release 6.3" "X Version 11"
.SH NAME
startx \- initialize an X session
.SH SYNOPSIS
.B startx
[ [
.I client
]
.I options
\&\.\|.\|. ] [
.B \-\^\-
[
.I server
]
.I options
\&.\|.\|. ]
.SH DESCRIPTION
NOTE: The \fIstartx\fP script supplied with the X11 distribution is a sample
designed more as a base for customization than as a
finished product.  Site administrators are urged to customize it for
their site.  And to update this manual page when they do!
.PP
The \fIstartx\fP script is a front end to \fIxinit\fP that provides a
somewhat nicer user interface for running a single session of the X
Window System.  It is typically run with no arguments.
.PP
To determine the client to run,
.I startx
first looks for a file called
.I .xinitrc
in the user's home directory.  If that is not found, it uses
the file
.I xinitrc
in the
.I xinit
library directory.
If command line client options are given, they override this
behavior.
To determine the server to run,
.I startx
first looks for a file called
.I .xserverrc
in the user's home directory.  If that is not found, it uses
the file
.I xserverrc
in the
.I xinit
library directory.
If command line server options are given, they override this
behavior.  Users rarely need to provide a
.I .xserverrc
file.
See the
.IR xinit (1)
manual page for more details on the arguments.
.PP
The
.I .xinitrc
is typically a shell script which starts many clients according to the
user's preference.  When this shell script exits,
.I startx 
kills the server and performs any other session shutdown needed.
Most of the clients started by
.I .xinitrc
should be run in the background.  The last client should run in the
foreground; when it exits, the session will exit.  People often choose
a session manager, window manager, or \fIxterm\fP as the ``magic'' client.
.SH EXAMPLE
.PP
Below is a sample \fI\.xinitrc\fP that starts several applications and
leaves the window manager running as the ``last'' application.  Assuming that
the window manager has been configured properly, the user
then chooses the ``Exit'' menu item to shut down X.
.sp
.in +4
.nf
xrdb  \-load  $HOME/.Xresources
xsetroot  \-solid  gray  &
xbiff  \-geometry  \-430+5  &
oclock  \-geometry  75x75\-0\-0  &
xload  \-geometry  \-80\-0  &
xterm  \-geometry  +0+60  \-ls  &
xterm  \-geometry  +0\-100  &
xconsole  \-geometry  \-0+0  \-fn  5x7  &
exec  twm
.fi
.in -4
.SH "ENVIRONMENT VARIABLES"
.TP 25
DISPLAY
This variable gets set to the name of the display to which clients should
connect.  Note that this gets
.I set,
not read.
.SH FILES
.TP 25
.I $(HOME)/.xinitrc
Client to run.  Typically a shell script which runs many programs in
the background.
.TP 25
.I $(HOME)/.xserverrc
Server to run.  The default is
.I X.
.TP 25
.I <XRoot>/lib/X11/xinit/xinitrc
Client to run if the user has no
.I .xinitrc
file.  <XRoot> refers to the root of the X11 install tree.
.TP 25
.I <XRoot>/lib/X11/xinit/xserverrc
Client to run if the user has no
.I .xserverrc
file.  This is only needed if the server needs special arguments or is
not named.  <XRoot> refers to the root of the X11 install tree.
.SH "SEE ALSO"
.IR xinit (1)


ace/ChangeLog
Fri Oct 05 05:56:50 2001  Balachandran  <bala@cs.wustl.edu>

        * ACE version 5.2 released.

Thu Oct  4 17:38:21 2001  Steve Huston  <shuston@riverace.com>

        * ace/Synch.h: Added some comments to explain the rationale for
          the different return values from the blocking and timed
          versions of ACE_Null_Semaphore::acquire().  Thanks to Doug
          Schmidt for explaining this.

Thu Oct  4 17:27:39 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Handler.cpp (recv_log_record): Insure that
          the payload message block is properly aligned for any CDR
          extraction. Many thanks to Bala for noticing this problem and
          sending code for the fix.

Thu Oct  4 17:08:39 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Process_Per_Connection_Logging_Server.cpp:
          In prepare(), call ACE_Process_Options::command_line() rather
          than ACE_Process_Options::process_name() to set the program
          path for the child process.

Thu Oct  4 13:18:37 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Handler.cpp: Use a ACE_Message_Block,
          rather than hijacking a CDR block, to read the payload into.

Thu Oct  4 10:14:34 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* THANKS: Added Donald Acton to the Hall of Fame.

Thu Oct  4 06:39:20 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* ace/config-sunos5.5.h: Added ACE_LACKS_MKSTEMP as SunOS 5.5 does
	  not support mkstemp (). 
	
	* ace/config-sunos5.6.h: Undefined ACE_LACKS_MKSTEMP as Sun 5.6
	  and above support it. 

	  Thanks to Don Hinton <dhinton@swan.fr> for reporting this
	  problem and Chad Elliott for testing this out on a Sun OS 5.5
	  box. 

Wed Oct  3 16:37:41 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* ace/CDR_Stream.cpp (read_1): Reverted this change "Wed Oct  3
	  15:05:43 2001  Jeff Parsons <parsons@cs.wustl.edu>".  This
	  change is alone not enough to fix the problem that Brad
	  reported. The sample program for the text needs to be fixed to
	  get it working right.

Wed Oct  3 15:05:43 2001  Jeff Parsons <parsons@cs.wustl.edu>

        * ace/CDR_Stream.cpp (read_1);

          Reverted changed made in 

          Fri Aug 17 16:56:16 2001  Jeff Parsons <parsons@cs.wustl.edu>

          to read_1(). A similar change made in several other methods
          is ok, but in this case bytes can be copied into the
          input CDR stream without changing wr_ptr, so the check will
          fail unnecessarily. Thanks to Brad Hoskins <BHoskins@slo.newport.com>
          for pointing this out.

Wed Oct  2 12:18:37 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Server.cpp: Fixed spelling error.
        * examples/C++NPv1/Logging_Handler.cpp: Comment length adjustment.

Tue Oct  2 21:52:31 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Client.cpp: Reverted this:
          Mon Oct  1 18:27:41 2001  Steve Huston  <shuston@riverace.com>

Tue Oct  2 19:52:57 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* Makefile: Removed some of the files that no longer exist in the
	  repository. The inclusion of these files in the Makefile added
	  noise during the beta cutting process.

Tue Oct 02 15:34:49 2001  Carlos O'Ryan  <coryan@uci.edu>

	* etc/tao_cosevent.doxygen:
	* etc/tao_cosnotification.doxygen:
	  Fixed typos in the tag file names.

Tue Oct  2 15:52:27 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* THANKS: Added Sunny Leung to the Hall of Fame.

Tue Oct  2 15:33:36 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* THANKS: Removed one instance of Peter Phillips
	  <pphillip@opentext.com> from the Hall of Fame.

Tue Oct  2 15:00:37 2001  Balachandran Natarajan  <bala@cs.wustl.edu>

	* ace/CDR_Stream.cpp (clone_from): Copied the GIOP major and minor
	  versions from the incoming CDR stream. Thanks to Sunny Leung 
	  <sleung@opentext.com> for reporting this. This closes bugzilla
	  #1043.  

Tue Oct  2 09:45:39 2001  Steve Huston  <shuston@riverace.com>

        * tests/libDLL_Test.icc:
        * tests/libService_Config_DLL.icc: Add link(exportAll) to the link
          options to export the entrypoints needed for the tests.

Mon Oct  1 20:04:14 2001  Priyanka Gontla  <pgontla@ece.uci.edu>

	* THANKS: 
	  Added Tushar Nair to the THANKS file.

Mon Oct  1 18:06:48 2001  Priyanka Gontla  <pgontla@ece.uci.edu>

	* ace/Object_Manager.cpp (ACE_UnhandledExceptionFilter):
	  Removed the un-needed #if 0's. Left them before as a safety
	  measure (in case something doesnt work) and forgot to remove
	  them later.

	* ace/Log_Msg.cpp (instance):
	  Fixed the seg faults that are raised when ace is compiled with
	  ACE_NTRACE=0 on SunOS. Added a check before locking and
	  releasing a lock.

	  Thanks to Tushar Nair <ntushar@yahoo.com> for reporting the
	  problem and for providing a fix.

Mon Oct  1 18:27:41 2001  Steve Huston  <shuston@riverace.com>

        * examples/C++NPv1/Logging_Client.cpp: Remove the third (optional)
          arg from getline(). Using a char literal without further
          qualification causes confusion in Unicode builds.

Mon Oct  1 14:44:13 2001  Ossama Othman  <ossama@uci.edu>

	* ace/SSL/SSL_Context.cpp (ssl_library_init):

	  Initialize the locking callbacks in the crypto library before
	  initializing the OpenSSL library.  OpenSSL may require the
	  locking callbacks to be initialized before its own
	  initialization.

	* ace/SSL/SSL_Context.cpp (ssl_library_fini):

	  Perform crypto library resource cleanup before reclaiming the
	  locks used by the locking callbacks.  Crypto library resource
	  cleanup makes use of the locking callbacks.

Mon Oct  1 15:19:11 2001  Douglas C. Schmidt  <schmidt@macarena.cs.wustl.edu>


XFree86-3.3/xc/programs/xinit/startx.cpp
XCOMM!/bin/sh

XCOMM $XConsortium: startx.cpp,v 1.4 91/08/22 11:41:29 rws Exp $
XCOMM $XFree86: xc/programs/xinit/startx.cpp,v 3.0 1994/05/22 00:02:28 dawes Exp $
XCOMM 
XCOMM This is just a sample implementation of a slightly less primitive 
XCOMM interface than xinit.  It looks for user .xinitrc and .xserverrc
XCOMM files, then system xinitrc and xserverrc files, else lets xinit choose
XCOMM its default.  The system xinitrc should probably do things like check
XCOMM for .Xresources files and merge them in, startup up a window manager,
XCOMM and pop a clock and serveral xterms.
XCOMM
XCOMM Site administrators are STRONGLY urged to write nicer versions.
XCOMM 

#ifdef SCO

XCOMM Check for /usr/bin/X11 and BINDIR in the path, if not add them.
XCOMM This allows startx to be placed in a place like /usr/bin or /usr/local/bin
XCOMM and people may use X without changing their PATH

XCOMM First our compiled path

bindir=BINDIR
if expr $PATH : ".*`echo $bindir | sed 's?/?\\/?g'`.*" > /dev/null 2>&1; then
	:
else
	PATH=$PATH:BINDIR
fi

XCOMM Now the "SCO" compiled path

if expr $PATH : '.*\/usr\/bin\/X11.*' > /dev/null 2>&1; then
	:
else
	PATH=$PATH:/usr/bin/X11
fi

XCOMM Set up the XMERGE env var so that dos merge is happy under X

if [ -f /usr/lib/merge/xmergeset.sh ]; then
	. /usr/lib/merge/xmergeset.sh
else if [ -f /usr/lib/merge/console.disp ]; then
	XMERGE=`cat /usr/lib/merge/console.disp`
	export XMERGE
fi
fi

scoclientrc=$HOME/.startxrc
#endif

userclientrc=$HOME/.xinitrc
userserverrc=$HOME/.xserverrc
sysclientrc=XINITDIR/xinitrc
sysserverrc=XINITDIR/xserverrc
clientargs=""
serverargs=""

#ifdef SCO
if [ -f $scoclientrc ]; then
    clientargs=$scoclientrc
else
#endif
if [ -f $userclientrc ]; then
    clientargs=$userclientrc
else if [ -f $sysclientrc ]; then
    clientargs=$sysclientrc
fi
fi
#ifdef SCO
fi
#endif

if [ -f $userserverrc ]; then
    serverargs=$userserverrc
else if [ -f $sysserverrc ]; then
    serverargs=$sysserverrc
fi
fi

whoseargs="client"
while [ "x$1" != "x" ]; do
    case "$1" in
	/''*|\.*)	if [ "$whoseargs" = "client" ]; then
		    clientargs="$1"
		else
		    serverargs="$1"
		fi ;;
	--)	whoseargs="server" ;;
	*)	if [ "$whoseargs" = "client" ]; then
		    clientargs="$clientargs $1"
		else
		    serverargs="$serverargs $1"
		fi ;;
    esac
    shift
done

xinit $clientargs -- $serverargs

/*
 * various machines need special cleaning up
 */
#ifdef macII
Xrepair
screenrestore
#endif

#ifdef sun
kbd_mode -a
#endif
